zstacklib/zstacklib/utils/file_metadata_handler.py

```
@@ -0,0 +1,273 @@
+            tmp_path = metadataPath + ".tmp"
+            if os.path.isfile(tmp_path):
+                try:
+                    with open(tmp_path, 'r') as f:
+                        content = f.read()
+                    json.loads(content)  # validate JSON completeness
+                    os.rename(tmp_path, metadataPath)
+                    logger.info("recovered tmp file to %s during get", metadataPath)
+                except (ValueError, TypeError):
+                    logger.warn("tmp file %s contains incomplete JSON", tmp_path)
+                    return {'metadata': None}
+                except Exception as e:
```

Ruff (0.15.6)

[warning] Do not catch blind exception: `Exception`

(BLE001)